### PR TITLE
PaySpace (Independent Publisher)

### PR DESCRIPTION
**Pull request by mint-virgil**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by mint-virgil** [comment]
@microsoft-github-policy-service agree [company="Mint Management Technologies"]

**Comment by mint-virgil** [comment]
@microsoft-github-policy-service agree company="Mint Management Technologies"

**Comment by Amjed-Ayoub** [comment]
@mint-virgil thank you so much for your new submission.
If you would like us to review your connector files, please remove the "Proposal" from the PR title.

Thank you so much.

**Comment by mint-virgil** [comment]
> @mint-virgil thank you so much for your new submission. If you would like us to review your connector files, please remove the "Proposal" from the PR title.
> 
> Thank you so much.

Thanks @Amjed-Ayoub - will do. 
I do have a question that I have noticed. I've added 2 custom policies to the connector but I don't see it in json file. Is there something specific that I have not done when I downloaded the file in the portal?

**Comment by Amjed-Ayoub** [comment]
> > @mint-virgil thank you so much for your new submission. If you would like us to review your connector files, please remove the "Proposal" from the PR title.
> > Thank you so much.
> 
> Thanks @Amjed-Ayoub - will do. I do have a question that I have noticed. I've added 2 custom policies to the connector but I don't see it in json file. Is there something specific that I have not done when I downloaded the file in the portal?

@mint-virgil you can add the policies templates in the apiProperties.json file. please see [here ](https://github.com/microsoft/PowerPlatformConnectors/pull/2332/files#diff-ef1558e405c640a0e894508a2e90d03d1448ec942c7875b4d274f37f0ff355d4:~:text=%22policyTemplateInstances%22)for an example.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,4476 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "PaySpace",
+    "description": "Our powerful API allows you to access your employee data in order to utilize in your business environment. All supported endpoints in one place and ready to be integrated into your solution.",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:6]: @mint-virgil Please remove the work "API" from the description. please follow our requirements [here ](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#give-your-connector-a-title)and [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#connector-info).

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,4476 @@
...
+    "contact": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:7]: @mint-virgil please have your information or your organization information in the contact section. It should include the publisher information.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,4476 @@
...
+            "name": "CustomEnvironmentHeader",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "Variable that is either api (for PROD) or apistaging (for STAGING)",
+            "x-ms-summary": "Environment Variable"
+          }
+        ],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:104]: @mint-virgil is not allowed to have a default response only. please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).
Please fix all responses in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,4476 @@
...
+            "description": "default",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:105]: @mint-virgil Please change to a proper description. for example you can use "OK", "Success", "Bad Request", ...etc.
Please fix all responses' descriptions in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,4476 @@
...
+            "name": "CustomEnvironmentHeader",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "Variable that is either api (for PROD) or apistaging (for STAGING)",
+            "x-ms-summary": "Environment Variable"
+          }
+        ],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3829]: @mint-virgil is not allowed to have a default response only. please define a proper response. please see here.
Please fix all responses in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,4476 @@
...
+            "description": "default",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3830]: @mint-virgil Please change to a proper description. for example you can use "OK", "Success", "Bad Request", ...etc.
Please fix all responses' descriptions in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,4476 @@
...
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3831]: @mint-virgil empty schema is not allowed. Please remove or define a proper schema.
Please fix all empty schemas in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,4476 @@
...
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {
+              "type": "object",
+              "properties": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3107]: @mint-virgil is it possible to remove the empty properties. if you are trying to use dynamic schema, please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/openapi-extensions#how-to-use-dynamic-schema).
Please fix all schemas in your connector swagger file.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,4476 @@
...
+  "security": [],
+  "tags": [],
+  "x-ms-connector-metadata":[
+    {
+        "propertyName": "Website",
+        "propertyValue": "https://www.payspace.com"
+    },
+    {
+        "propertyName": "Privacy policy",
+        "propertyValue": "https://www.payspace.com/privacy-policy/;https://za.mintgroup.net/privacy-policy/"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:4469]: @mint-virgil please include the stackowner privacy policy page only. Thank you

Comment (Amjed-Ayoub) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:4469]: You can add yours in the readme file :-).

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/PaySpace/README.md [hunk 10]
```diff
@@ -0,0 +1,361 @@
+
```
[the remark this answers is not in the document]
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/README.md:1]: Thanks @Amjed-Ayoub - I believe I've made the necessary changes as per the example.

File: independent-publisher-connectors/PaySpace/README.md [hunk 11]
```diff
@@ -0,0 +1,361 @@
...
+  
+
+  
+
+## Positions
+
+  
+
+Retrieve employees' postion related information associated with the company-id.
```
[the remark this answers is not in the document]
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/README.md:52]: Thanks @Amjed-Ayoub - I've removed these spacings

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 1 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: @Amjed-Ayoub noted on this item

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 2 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:7]: @Amjed-Ayoub I've put in my organization information in here now.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 9 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:4469]: @Amjed-Ayoub - noted on the responses - I'll look into that now.
Thank you very much

File: independent-publisher-connectors/PaySpace/README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/PaySpace/README.md:51]: @mint-virgil here you havea typo "...postion...", can you please change it to "...position...".

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 2 above)
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:7]: the "name" and the "email" Properties are still for "PaySpace". can you please replace with your information or your organization information?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
@mint-virgil Please resolve all conversations and comments in this PR. Thank you so much for your hard work with us.

**Comment by Amjed-Ayoub** [comment]
> > > @mint-virgil thank you so much for your new submission. If you would like us to review your connector files, please remove the "Proposal" from the PR title.
> > > Thank you so much.
> > 
> > 
> > Thanks @Amjed-Ayoub - will do. I do have a question that I have noticed. I've added 2 custom policies to the connector but I don't see it in json file. Is there something specific that I have not done when I downloaded the file in the portal?
> 
> @mint-virgil you can add the policies templates in the apiProperties.json file. please see [here ](https://github.com/microsoft/PowerPlatformConnectors/pull/2332/files#diff-ef1558e405c640a0e894508a2e90d03d1448ec942c7875b4d274f37f0ff355d4:~:text=%22policyTemplateInstances%22)for an example.

@mint-virgil I would like to provide this doc for [Policy templates](https://learn.microsoft.com/en-us/connectors/custom-connectors/policy-templates).

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,4476 @@
...
+            "type": "string",
+            "description": "{{company-id}} The Company Id",
+            "x-ms-summary": "{{company-id}}"
+          },
+          {
+            "name": "EmployeeId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "{{EmployeeId}} The EmployeeId"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:1315]: @mint-virgil The 'x-ms-summary' property is required. please add.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,4476 @@
...
+            "description": " Filter by field eg. \"EmployeeNumber eq 'Emp01'\"",
+            "x-ms-summary": "Optional $filter"
+          },
+          {
+            "name": "$expand",
+            "default": "OrganizationGroups",
+            "in": "query",
+            "type": "string",
+            "required": true,
+            "x-ms-summary": "$expand"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:2310]: @mint-virgil The 'description' property is required.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,4476 @@
...
+        "summary": "Update a single position record",
+        "description": "Update a single employee position record based on the specified EmployeePositionId.",
+        "operationId": "UpdateASinglePositionRecord",
+        "parameters": [
+          {
+            "name": "EmployeePositionId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "{{EmployeePositionId}} The EmployeePositionId"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3067]: @mint-virgil The 'x-ms-summary' property is required.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json [hunk 15]
```diff
@@ -0,0 +1,4476 @@
...
+            "description": " (true/false) - Return total count in response",
+            "x-ms-summary": "Optional $count"
+          },
+          {
+            "name": "$expand",
+            "default": "OrganizationGroups",
+            "in": "query",
+            "type": "string",
+            "required": true,
+            "x-ms-summary": "Optional $expand"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3157]: @mint-virgil The 'description' property is required.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]
@mint-virgil please resolve all comments and conversations. Thank you so much.

File: independent-publisher-connectors/PaySpace/README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/README.md]: Hi there, I've updated this type in the latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 2 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:7]: Hi there, I've updated these details in the latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 3 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: @Amjed-Ayoub, I am still busy with the responses - will commit again shortly to rectify them

Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: Hi @Amjed-Ayoub I've made the required changes to the responses. Kindly advise if they're correct now.

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 5 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 6 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 7 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 8 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 9 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 12 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in the latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 13 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:2334]: updated in the latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 14 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: updated in the latest commit

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 15 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json:3190]: updated in latest commit

File: independent-publisher-connectors/PaySpace/README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/README.md]: Updated in the latest commit @Amjed-Ayoub

File: independent-publisher-connectors/PaySpace/apiDefinition.swagger.json (hunk 4 above)
Comment (mint-virgil) [reply independent-publisher-connectors/PaySpace/apiDefinition.swagger.json]: resolved in previous commit

**Review by mint-virgil — COMMENTED** [review]
Hi @Amjed-Ayoub I believe I've made all the requested updated. Kindly confirm. Thank You

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @mint-virgil,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by mint-virgil** [comment]
> Hello @mint-virgil, I hope you are doing well. Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. Please let us know if you have any questions. Thank you very much for working with us.

Thank You @Amjed-Ayoub :)

**Comment by Amjed-Ayoub** [comment]
Hello @mint-virgil,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @mint-virgil,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Pull request merged**